pkg/kubelet/cm/event_dispatcher.go

```diff
@@ -76,10 +76,10 @@ func newEventDispatcher() *eventDispatcher {
 	return dispatcher
 }
 
-func (ed *eventDispatcher) dispatchEvent(pod *api.Pod, cgroupPath string, kind lifecycle.Event_Kind) error {
+func (ed *eventDispatcher) dispatchEvent(pod *api.Pod, cgroupPath string, kind lifecycle.Event_Kind) ([]*lifecycle.EventReply, error ){
```

Comment:
I think it depends whether Pod should fail when preStartHook will fail. Right now we are returning immediately when reply has Error field and do not continue iterating through handler list.